contracts/mixnet/src/mixnodes/transactions.rs

```diff
@@ -204,6 +204,40 @@ pub(crate) fn _try_remove_mixnode(
     Ok(response)
 }
 
+pub(crate) fn try_update_mixnode_config(
+    deps: DepsMut,
+    info: MessageInfo,
+    profit_margin_percent: u8,
+) -> Result<Response, ContractError> {
+    let owner = deps.api.addr_validate(info.sender.as_ref())?;
+    let mix_identity = storage::mixnodes()
+        .idx
+        .owner
+        .item(deps.storage, owner.clone())?
+        .ok_or(ContractError::NoAssociatedMixNodeBond { owner })?
+        .1
+        .identity()
+        .clone();
+
+    // We don't have to check lower bound as its an u8
+    if profit_margin_percent > 100 {
+        return Err(ContractError::InvalidProfitMarginPercent(
+            profit_margin_percent,
+        ));
+    }
+
+    storage::mixnodes().update(deps.storage, &mix_identity, |mixnode_bond_opt| {
+        mixnode_bond_opt
+            .map(|mut mixnode_bond| {
```

Comment:
Shouldn't we also update the mixnode `bond_height` so that somebody wouldn't attempt to change their profit margin just before rewards are distributed and thus scam all of their delegators?

Reply:
I think you're right, I'll modify that as well.